### PR TITLE
Terraform Example Code Optimimzed

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by marifse** [comment]
> @marifse After removing defaults the example requires mandatory values, which means now you need to update the `cra-config.yaml` accordingly to make it pass the pipeline.

@rajatagarwal-ibm the values added to cra-config.yaml, can you please check now

File: examples/backup/version.tf [hunk 1]
```diff
@@ -1,7 +1,8 @@
 terraform {
-  required_version = ">= 1.3.0, <1.7.0"
+  required_version = ">= 1.3.0, <1.6.0"
```
[the remark this answers is not in the document]
Comment (marifse) [reply examples/backup/version.tf]: @ocofaigh please check now, all fixed

File: tests/other_test.go [hunk 2]
```diff
@@ -64,36 +62,28 @@ func TestRunBasicExample(t *testing.T) {
...
-
-func TestPlanICDVersions(t *testing.T) {
-	t.Parallel()
-
-	// This test will run a terraform plan on available stable versions of postgresql
-	versions, _ := sharedInfoSvc.GetAvailableIcdVersions("postgresql")
-	for _, version := range versions {
-		t.Run(version, func(t *testing.T) { testPlanICDVersions(t, version) })
-	}
-}
```
[the remark this answers is not in the document]
Comment (marifse) [reply tests/other_test.go:99]: @ocofaigh please verify, All fixed

**Comment by marifse** [comment]
> left some comments

@ocofaigh fix have been applied, can you kindly recheck. thanks

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**